integration/components/groups/GroupDetail.tsx

```diff
@@ -133,6 +137,40 @@ export default class GroupDetail extends React.Component<GroupDetailProps, Group
         return null;
     }
 
+    /**
+     * Verify the group ID the user typed is correct before making the actual delete call.
+     */
+    roatateGroupPrivateKey = () => {
+        IronWeb.group
+            .rotatePrivateKey(this.props.group.groupID)
+            .then((groupKeyUpdate) => {
+                logAction(`Group private key rotation was successful`);
+                this.setState({needsRotation: groupKeyUpdate.needsRotation});
+                this.getRotateGroupPrivateKeyIcon();
```

Comment:
There's no need to call this method here. It's not doing anything. With React, the `render()` function automatically gets called after you call `this.setState` so that the UI reflects the new state.